packages/bsky/src/api/app/bsky/feed/getTimeline.ts

```diff
@@ -5,27 +5,40 @@ import { paginate } from '../../../../db/pagination'
 import AppContext from '../../../../context'
 import Database from '../../../../db'
 import { SkeletonFeedPost } from '../../../../lexicon/types/app/bsky/feed/defs'
+import { isView } from '../../../../lexicon/types/app/bsky/embed/record'
+import { PostTypeSet } from '../util/postTypeSet'
 
 export default function (server: Server, ctx: AppContext) {
   server.app.bsky.feed.getTimeline({
     auth: ctx.authVerifier,
     handler: async ({ params, auth }) => {
-      const { algorithm, limit, cursor } = params
+      const { algorithm, excludePostTypes, limit, cursor } = params
       const viewer = auth.credentials.did
 
       if (algorithm && algorithm !== FeedAlgorithm.ReverseChronological) {
         throw new InvalidRequestError(`Unsupported algorithm: ${algorithm}`)
       }
 
-      const skeleton = await getTimelineSkeleton(ctx.db, viewer, limit, cursor)
+      const skeleton = await getTimelineSkeleton(
+        ctx.db,
+        viewer,
+        limit,
+        excludePostTypes,
+        cursor,
+      )
 
       const feedService = ctx.services.feed(ctx.db)
       const feedItems = await feedService.cleanFeedSkeleton(
         skeleton.feed,
         limit,
         viewer,
       )
-      const feed = await feedService.hydrateFeed(feedItems, viewer)
+      let feed = await feedService.hydrateFeed(feedItems, viewer)
+
+      const excludePostTypeSet = new PostTypeSet(excludePostTypes)
```

Comment:
Had to do quote exclusion after hydration as I could not figure out how to add it to the query builder.